app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt

```diff
@@ -199,7 +207,23 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
                 ) {
                     updateSyncEngineState(context, SyncEngine.Passwords, newValue as Boolean)
                 } else {
-                    showPinDialogWarning(newValue as Boolean)
+                    showPinDialogWarning(SyncEngine.Passwords, newValue as Boolean)
+                }
+                true
+            }
+        }
+
+        requirePreference<CheckBoxPreference>(SyncEngine.CreditCards.prefId()).apply {
+            setOnPreferenceChangeListener { _, newValue ->
+                val manager =
+                    activity?.getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
+                if (manager.isKeyguardSecure ||
+                    newValue == false ||
+                    !context.settings().shouldShowSecurityPinWarningSync
+                ) {
+                    updateSyncEngineState(context, SyncEngine.CreditCards, newValue as Boolean)
+                } else {
+                    showPinDialogWarning(SyncEngine.CreditCards, newValue as Boolean)
                 }
```

Comment:
Can we refactor this out to a helper function that can be called by both with the SyncEngine as a param?

Reply:
Yes! good idea